actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -346,7 +346,7 @@ def check_part(name, part, path_params, hash)
           end
 
           def split_to(to)
-            if /#/.match?(to)
+            if !to.nil? && /#/.match?(to)
```

Comment:
I would simplify to `if to && /#/.match?(to)` here and other places.
If `to` was `false`, it would already raise a TypeError in <=2.6.